scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -531,10 +531,16 @@ object Execution {
    */
   private case class FlowDefExecution(result: (Config, Mode) => FlowDef) extends Execution[Unit] {
     protected def runStats(conf: Config, mode: Mode, cache: EvalCache)(implicit cec: ConcurrentExecutionContext) = {
-      lazy val future = cache
-        .runner
-        .validateAndRun(conf, mode)(result)
-        .map { m => ((), m) }
+      lazy val future = {
+        cache.writer match {
+          case ar: AsyncFlowDefRunner =>
+            ar.validateAndRun(conf, mode)(result).map { m => ((), Map(m)) }
+          case other =>
```

Comment:
what Ian said. The goal here is to make hopefully all of the typed API not depend on cascading and just make the cascading backend one implementation. I think no one uses this function anyway (creating an Execution from a FlowDef) and I want to just remove it, but in the mean time, I want to not dig us in any deeper.

If no one at Twitter is using it, I would suggest we remove the function, but keeping it around for 1 version deprecated may work okay too.